course_flow/tasks.py

```diff
@@ -0,0 +1,314 @@
+    if len(nodes) > 0:
+        df = df.append(
+            {
+                "a": _("Term"),
+                "b": WeekWorkflow.objects.get(week__nodes=nodes[0]).rank + 1,
+            },
+            ignore_index=True,
+        )
+        prereqs = Node.objects.filter(
+            outgoing_links__target_node__in=nodes
+        ).distinct()
+        postreqs = Node.objects.filter(
+            incoming_links__source_node__in=nodes
+        ).distinct()
+        if len(prereqs) > 0:
+            df = df.append(
+                {
+                    "a": _("Prerequisites"),
+                    "b": [get_displayed_title(req) for req in prereqs].join(
+                        ", "
+                    ),
+                },
+                ignore_index=True,
+            )
+        if len(postreqs) > 0:
+            df = df.append(
+                {
+                    "a": _("Required For"),
+                    "b": [get_displayed_title(req) for req in postreqs].join(
+                        ", "
+                    ),
+                },
+                ignore_index=True,
+            )
+    df = df.append(
+        {
+            "a": _("Course Outcome"),
+            "b": _("Sub-Outcomes"),
+            "c": _("Competencies"),
+            "d": _("Topics & Content"),
+            "e": _("Lessons/Activities"),
+            "f": _("Assessments"),
+        },
+        ignore_index=True,
+    )
+    for outcome in workflow.outcomes.all():
+        df = df.append(
+            get_framework_line_for_outcome(outcome), ignore_index=True
+        )
+    return df
+
+
+def get_workflow_outcomes_table(workflow):
+    outcomes = get_all_outcomes_ordered(workflow)
+    data = OutcomeExportSerializer(outcomes, many=True).data
+    df = pd.DataFrame(
+        data, columns=["code", "title", "description", "id", "depth"]
+    )
+    pd.set_option("display.max_colwidth", None)
+    return df
+
+
+@shared_task
+def async_get_outcomes_excel(email, pk, object_type):
+    model_object = get_model_from_str(object_type).objects.get(pk=pk)
+    with BytesIO() as b:
+        # Use the StringIO object as the filehandle.
+        writer = pd.ExcelWriter(b, engine="openpyxl")
+        if object_type == "workflow":
+            workflows = [model_object]
+        elif object_type == "project":
+            workflows = list(model_object.workflows.all())
+        for workflow in workflows:
+            df = get_workflow_outcomes_table(workflow)
+            df.to_excel(
+                writer,
+                sheet_name=workflow.title + "_" + str(workflow.pk),
+                index=False,
+            )
+            writer.save()
+        # Set up the Http response.
+        filename = (
+            object_type
+            + "_"
+            + str(pk)
+            + "_"
+            + timezone.now().strftime(dateTimeFormatNoSpace())
+            + ".xlsx"
+        )
+        email = EmailMessage(
+            "Your Outcomes Export",

```

Comment:
This needs to be marked for translation as well.  Since this happens outside of a request however I think it will always use default language.

Reply:
It might be better to build the `EmailMessage` object in the view and pass the whole thing in.

Reply:
Passing the `EmailMessage` object is tricky because it's not JSON serializable. I think it's easier to just build it here.

Reply:
Responding here because the email replies are a bit clunky: I added the gettext when I moved them to the views then forgot to replace it when I moved this back to tasks.py. I've just pushed a new commit where that is fixed. My bad.